src/Dropdown/Dropdown.tsx

```diff
@@ -22,30 +22,29 @@ export type NonDefaultProps<OptionType> = CommonProps<OptionType> &
         isClearable?: false;
         value: OptionType;
         onChange: (value: OptionType) => void;
-      });
+      }
+  );
 
 export namespace Dropdown {
   export type Props<OptionType> = NonDefaultProps<OptionType> &
     Partial<DefaultProps>;
 }
 
 export class Dropdown<OptionType> extends React.PureComponent<
-  NonDefaultProps<OptionType> & DefaultProps
+  Dropdown.Props<OptionType>
```

Comment:
this was plain wrong before, requiring all the `DefaultProps` to be passed. Or am I missing something?

Reply:
If I remember correctly, Typescript3 should have fixed the use of `defaultProps` in react components, so having a static `defaultProps` should work fine now. Am I wrong?

Reply:
Could this be a good chance to use a function component?

```ts
function DropDown<OptionType>({ className, components: customComponents, innerRef, allowCreate = false, size, flat = false, ...props }: Props<OptionType> { ... }
```

Reply:
@gabro I don't see that too much related. If we want to switch to functional components we should do it in a separate PR for every component in the library.

Reply:
Nevermind, I got confused